csrc/multidevice/communicator.cpp

```diff
@@ -267,4 +267,13 @@ c10d::Backend* Communicator::getWorld(
   return getBackendForTeam(all_ranks, backend);
 }
 
+void Communicator::barrier(std::optional<CommunicatorBackend> backend) {
+  // Explicitly specify the device ID to avoid a warning. Without this,
+  // ProcessGroupNCCL::barrier may guess the wrong mapping and failed to block
+  // CPU properly:
+  // https://github.com/pytorch/pytorch/blob/7e4329c258306cc14303895e5f1e6036b009e74f/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L3905-L3912.
+  c10d::BarrierOptions options{.device_ids = {deviceId()}};
```

Comment:
Should we be using the local rank? It looks like their guessing function uses local rank, but they mention it also might not be right....
https://github.com/pytorch/pytorch/blob/7e4329c258306cc14303895e5f1e6036b009e74f/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L428
We might want to test this on two nodes to make certain.